module/product/src/Application/Controller/Api/Attribute/DeleteProductAttributeAction.php

```diff
@@ -0,0 +1,102 @@
+<?php
+/**
+ * Copyright © Bold Brand Commerce Sp. z o.o. All rights reserved.
+ * See LICENSE.txt for license details.
+ */
+
+declare(strict_types=1);
+
+namespace Ergonode\Product\Application\Controller\Api\Attribute;
+
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\IsGranted;
+use Swagger\Annotations as SWG;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\Annotation\Route;
+use Ergonode\Product\Domain\Entity\AbstractProduct;
+use Ergonode\Core\Domain\ValueObject\Language;
+use Ergonode\Attribute\Domain\Entity\AbstractAttribute;
+use Symfony\Component\HttpKernel\Exception\AccessDeniedHttpException;
+use Ergonode\EventSourcing\Infrastructure\Bus\CommandBusInterface;
+use Ergonode\Core\Domain\Query\LanguageQueryInterface;
+use Ergonode\Product\Domain\Command\Attribute\RemoveProductAttributeCommand;
+use Ergonode\Api\Application\Response\EmptyResponse;
+
+class DeleteProductAttributeAction
+{
+    private CommandBusInterface $commandBus;
+
+    private LanguageQueryInterface $query;
+
+    public function __construct(CommandBusInterface $commandBus, LanguageQueryInterface $query)
+    {
+        $this->commandBus = $commandBus;
+        $this->query = $query;
+    }
+
+    /**
+     * @Route(
+     *     name="ergonode_product_attribute_delete",
+     *     path="products/{product}/attribute/{attribute}",
+     *     methods={"DELETE"},
+     *     requirements={
+     *         "product"="[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}",
+     *         "attribute"="[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}"
+     *     }
+     * )
+     * @Route(
+     *     "/products/{product}/draft/{attribute}/value",
+     *     methods={"DELETE"},
+     *     requirements = {
+     *        "product" = "[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}",
+     *        "attribute" = "[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}"
+     *     }
+     * )
+     *
+     * @IsGranted("PRODUCT_UPDATE")
+     * @IsGranted("edit", subject="language")
+     *
+     * @SWG\Tag(name="Product")
+     * @SWG\Parameter(
+     *     name="product",
+     *     in="path",
+     *     type="string",
+     *     description="Product id",
+     * )
+     * @SWG\Parameter(
+     *     name="attribute",
+     *     in="path",
+     *     type="string",
+     *     description="Attribute id",
+     * )
+     * @SWG\Parameter(
+     *     name="language",
+     *     in="path",
+     *     type="string",
+     *     required=true,
+     *     default="en_GB",
+     *     description="Language Code",
+     * )
+     * @SWG\Response(
+     *     response=200,
+     *     description="Change product attribute Value",
+     * )
+     *
+     * @throws \Exception
+     */
+    public function removeDraftAttribute(
+        AbstractProduct $product,
+        Language $language,
+        AbstractAttribute $attribute
+    ): Response {
+        if ($attribute->getScope()->isGlobal()) {
```

Comment:
It seems like a generic rule. Wouldn't be worth moving to param converter?

Reply:
Is on end of refactoring, not shour is this Action stay on code, now is required for compatibility